sound/soc/sof/Kconfig

```diff
@@ -186,6 +186,7 @@ endif ## SND_SOC_SOF_DEVELOPER_SUPPORT
 
 config SND_SOC_SOF
 	tristate
+	select VIRTUAL_BUS
```

Comment:
is client support in SOF going to be optional or compulsory? If compulsory, why wouldn't "select" be a good option?

Reply:
@lyakh select is bad here because it should not be used with user-visible options (which VIRTUAL_BUS is). But client support in general cannot be optional if we want to support audio as a separate client. So I am not sure about what do do with it

Reply:
I personally view the multi-client as an option. If someone has no need for it we should be able to disable it and have basic audio working, no?

Reply:
@plbossart OK that makes sense and it also makes the implementation a bit tricky but thats my problem haha. Let me update this series with the suggested changes in the kconfig

Reply:
It's harder to 'think lean' with minimal options, but in general it forces engineers to come-up with good/scalable designs :-)

Reply:
btw the one rule is: don't use select when the option depends on something else. Select will ignore the dependencies and this results in broken builds.

To either put the depends at a higher level, or use a select XYZ if <dependencies>

There's also an indirect what to do this with booleans that are used to conditionally select stuff, that's what we did for PCI/ACPI stuff with all the _SUPPORT options.

Reply:
Ahh good to know @plbossart . VIRTUAL_BUS has no depencies. So it is safe to just use select here?

Reply:
you could select it from the CLIENT option then